pkg/api/validation/schema.go

```diff
@@ -147,8 +148,15 @@ func (s *SwaggerSchema) ValidateObject(obj interface{}, apiVersion, fieldName, t
 	return allErrs
 }
 
+var versionRegexp = regexp.MustCompile(`^v.+\..*`)
+
 func (s *SwaggerSchema) validateField(value interface{}, apiVersion, fieldName, fieldType string, fieldDetails *swagger.ModelProperty) errs.ValidationErrorList {
-	if strings.HasPrefix(fieldType, apiVersion) {
+	// TODO: caesarxuchao: because we have multiple group/versions and objects
+	// may reference objects in other group, the commented out way of checking
+	// if a filedType is a type defined by us is outdated. We use a hacky way
+	// for now.
```

Comment:
like this https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/experimental/v1/types.go#L205

and during the validation, the old mechanism will check `strings.HasPrefix(v1.PodTemplateSpec, v1alpha1)`, so it will return a false. But we should take this if condition and recursively cal ValidateObject as v1.PodTemplateSpec is a type defined by us.

Do you mean I should give an example in the comments?